third_party/move/move-compiler-v2/tests/checking/typing/eq_inline_typed.move

```diff
@@ -0,0 +1,13 @@
+module 0x42::m {
+
```

Comment:
The referenced issue #6922 says "type annotations are required to help the inference algorithm in some occasions". Do we have an example/test case where the type inference was not able to infer the types when lambda params were not explicitly typed, but the addition of the feature in this PR allows the user to progress in the natural way by adding explicit types (instead of the hack mentioned in the referenced issue for explicit typing)?

Reply:
I'm still trying to find such a case.